src/common/mfu_daos.c

```diff
@@ -610,6 +610,291 @@ static int daos_any_error(int rank, bool local_daos_error, int flag_daos_args)
     return 0;
 }
 
+/*
+ * Get the user attributes for a container in a format similar
+ * to what daos_cont_set_prop expects.
+ * The last entry is the ACL and is conditionally set only if
+ * the user has permissions.
+ * Returns 1 on error, 0 on success.
+ */
+static int cont_get_props(daos_handle_t coh, daos_prop_t** _props)
+{
+    int             rc;
+    daos_prop_t*    props = NULL;
+    daos_prop_t*    prop_acl = NULL;
+    daos_prop_t*    props_merged = NULL;
+
+    /* Allocate space for all props except ACL. */
+    props = daos_prop_alloc(16);
+    if (props == NULL) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to allocate container properties.");
+        rc = 1;
+        goto out;
+    }
+
+    /* The order of properties MUST match the order expected by serialization. */
+    props->dpp_entries[0].dpe_type = DAOS_PROP_CO_LABEL;
+
+    props->dpp_entries[1].dpe_type = DAOS_PROP_CO_LAYOUT_TYPE;
+    props->dpp_entries[2].dpe_type = DAOS_PROP_CO_LAYOUT_VER;
+    props->dpp_entries[3].dpe_type = DAOS_PROP_CO_CSUM;
+    props->dpp_entries[4].dpe_type = DAOS_PROP_CO_CSUM_CHUNK_SIZE;
+    props->dpp_entries[5].dpe_type = DAOS_PROP_CO_CSUM_SERVER_VERIFY;
+    props->dpp_entries[6].dpe_type = DAOS_PROP_CO_REDUN_FAC;
+    props->dpp_entries[7].dpe_type = DAOS_PROP_CO_REDUN_LVL;
+    props->dpp_entries[8].dpe_type = DAOS_PROP_CO_SNAPSHOT_MAX;
+    props->dpp_entries[9].dpe_type = DAOS_PROP_CO_COMPRESS;
+    props->dpp_entries[10].dpe_type = DAOS_PROP_CO_ENCRYPT;
+    props->dpp_entries[11].dpe_type = DAOS_PROP_CO_OWNER;
+    props->dpp_entries[12].dpe_type = DAOS_PROP_CO_OWNER_GROUP;
+    props->dpp_entries[13].dpe_type = DAOS_PROP_CO_DEDUP;
+    props->dpp_entries[14].dpe_type = DAOS_PROP_CO_DEDUP_THRESHOLD;
+    props->dpp_entries[15].dpe_type = DAOS_PROP_CO_ALLOCED_OID;
```

Comment:
@daltonbohning this property should not be set for POSIX containers.

Reply:
@dsikich Thanks! Specifically, it should not be set when we use DFS. I think we should still set it for POSIX -> POSIX with the Object API, just not DFS. It seemed the simplest way was to just pass a `get_oid` flag

Reply:
> @dsikich Thanks! Specifically, it should not be set when we use DFS. I think we should still set it for POSIX -> POSIX with the Object API, just not DFS. It seemed the simplest way was to just pass a `get_oid` flag

@daltonbohning yeah, I agree 👍